docs/Training-Imitation-Learning.md

```diff
@@ -1,3 +1,34 @@
 # Imitation Learning
 
-This feature and its associated documentation are still in development.
\ No newline at end of file
+It is often more intuitive to simply demonstrate the behavior we want an agent to perform, rather than attempting to have it learn via trial-and-error methods. Consider our [running example](ML-Agents-Overview.md#running-example-training-npc-behaviors) of training a medic NPC : instead of indirectly training a medic with the help of a reward function, we can give the medic real world examples of observations from the game and actions from a game controller to guide the medic's behavior. More specifically, in this mode, the Brain type during training is set to Player and all the actions performed with the controller (in addition to the agent observations) will be recorded and sent to the Python API. The imitation learning algorithm will then use these pairs of observations and actions from the human player to learn a policy.
+
+## Using Behavioral Cloning
+
+There are a variety of possible imitation learning algorithms which can be used, and for v0.3 we are starting with the simplest one: Behavioral Cloning. It works by collecting training data from a teacher, and then simply uses it to directly learn a policy, in the same way the supervised learning for image classification or other traditional Machine Learning tasks work.
+
+1. In order to use Behavioral Cloning in a scene, the first thing you will need is to create two Brains, one which will be the "Teacher," and the other which will be the "Student." We will assume that the names of the brain `GameObject`s are "Teacher" and "Student" respectively.
+2. Set the "Teacher" brain to Player mode, and properly configure the inputs to map to the corresponding actions. **Ensure that "Broadcast" is checked within the Brain inspector window.** 
+3. Set the "Student" brain to External mode.
+4. Link the brains to the desired agents (one agent as the teacher and at least one agent as a student).
+5. Build the Unity executable for your desired platform.
+6. In `trainer_config.yaml`,  add an entry for the "Student" brain. Set the `trainer` parameter of this entry to `imitation`, and the `brain_to_imitate` parameter to the name of the teacher brain: "Teacher". Additionally, set `batches_per_epoch`, which controls how much training to do each moment. Increase the `max_steps` option if you'd like to keep training the agents for a longer period of time.
+7. Launch the training process with `python3 python/learn.py <env_name> --train --slow`, where `<env_Name>` is the path to your built Unity executable.
```

Comment:
- We mix python 2 and 3 semantics throughout the doc ... use python[3] (?)
- Replace <env_Name> with <env_name>

Reply:
I am thinking we just stick to `python`?